plugins/rikki/heroeslounge/models/SlothTeamPivot.php

```diff
@@ -0,0 +1,51 @@
+<?php namespace Rikki\Heroeslounge\Models;
+
+use October\Rain\Database\Pivot;
+use Rikki\Heroeslounge\Models\Sloth;
+use Rikki\Heroeslounge\Models\Team;
+
+/**
+ * Sloth-Team Pivot Model
+ */
+class SlothTeamPivot extends Pivot
+{
+    
+    /*
+     * Disable timestamps by default.
+     * Remove this line if timestamps are defined in the database table.
+     */
+    public $timestamps = false;
+
+    /**
+     * @var string The database table used by the model.
+     */
+    public $table = 'rikki_heroeslounge_sloth_team';
+
+    public function afterSave()
+    {
+        if ($this->isDirty('is_captain')) {
+            $sloth = Sloth::find($this->sloth_id);            
+
+            if ($this->is_captain) {
+                $sloth->addDiscordCaptainRole();
+            } else {
+                // We need to check if the sloth is not captain of another team before syncing Discord.
+                if(!isCaptainOtherTeam($sloth, $this->team_id)) {
+                    $sloth->removeDiscordCaptainRole();
+                }
+            }
+        }
+    }
+
+    public function isCaptainOtherTeam($sloth, $updatedTeamId) {
+        $teamCaptaining = $sloth->teams->first(function($index, $team) {
+            return $team->id != $updatedTeamId && $team->pivot->is_captain;
+        });
+
+        if ($teamCaptaining != null) {
+            return true
+        }
+        return false;
```

Comment:
return true is missing a ;
In any case, I think this can be boiled down to return $sloth->... != null.
Probably better to use filter instead of first and then count() >= 1 instead of comparing with null, this should make the code even more readable.
I think the method itself should be named isCaptainOfOtherTeam or leadsOtherTeam